Library/Homebrew/cmd/fetch.rb

```diff
@@ -86,6 +86,15 @@ def fetch
       when Formula
         f = formula_or_cask
 
+        fetch_and_audit_deps = f.deps.select(&:fetch_and_audit?)
+        fetch_and_audit_deps_installed = fetch_and_audit_deps.select(&:installed?)
+
+        fetch_and_audit_deps.map do |dep|
+          next if fetch_and_audit_deps_installed.include?(dep)
+
+          safe_system ENV["HOMEBREW_BREW_FILE"], "install", dep.name
```

Comment:
I don't think we should be installing here. I think https://github.com/Homebrew/brew/blob/239bd3cdf79c4e60ddbeb839ddcc53a1837570bc/Library/Homebrew/dependency_collector.rb#L139-L144 

is the right place for this logic as it's what already handles auto-installing these sort of dependencies based on URLs.

Annoyingly, this may also mean that the approach may make more sense for this to be an attribute on the URL e.g. `using: NewCurlDownloadStrategy` or something.

I'm sorry to have not mentioned that earlier, I didn't think about that code until I saw this here.

Reply:
I had a look at this. It is possible to add a `BrewCurlDownLoadStrategy` that uses brew curl. But you need to pass a flag through the whole code down to this place: the entry point being https://github.com/Homebrew/brew/blob/cd900ef71eeecb621faba88aabc641fe93325fe6/Library/Homebrew/resource.rb#L176, i.e. the `url` method from `resource` blocks. This is where it gets tricky, as resources will now have to know if they should used brewed curl or not, based on the formula's `fetch and audit` dependencies. From the first analysis I made this looks like a bigger refactoring or change.

I will make an alternative proposal: the code in `fetch.rb` will install brewed curl based in the `fetch_and_audit` dependency, as already done in this PR. But the final curl call in `curl.rb` can be fixed directly for our issue. I'll push a new version of the PR.

Reply:
Why can't resources that need brewed `curl` also do `using: :brewed_curl` as with any other download strategy?

It might work for the `homepage` too...

Reply:
> This is where it gets tricky, as resources will now have to know if they should used brewed curl or not, based on the formula's `fetch and audit` dependencies.

I'm proposing this could eliminate "fetch and audit" dependencies entirely. I'm sorry for the suggested change in approach.

> Why can't resources that need brewed `curl` also do `using: :brewed_curl` as with any other download strategy?

Yeh, this.

> It might work for the `homepage` too...

This would be good, too.

Reply:
I see. I’m working on a solution with :using. Still need to connect some dots but it should work. I’ll update the PR accordingly